Lib/test/test_math.py

```diff
@@ -1899,14 +1899,22 @@ def testTan(self):
         self.ftest('tan(0)', math.tan(0), 0)
         self.ftest('tan(pi/4)', math.tan(math.pi/4), 1)
         self.ftest('tan(-pi/4)', math.tan(-math.pi/4), -1)
-        try:
-            self.assertTrue(math.isnan(math.tan(INF)))
-            self.assertTrue(math.isnan(math.tan(NINF)))
-        except:
-            self.assertRaises(ValueError, math.tan, INF)
-            self.assertRaises(ValueError, math.tan, NINF)
         self.assertTrue(math.isnan(math.tan(NAN)))
 
+        try:
+            tan = math.tan(INF)
+        except ValueError:
+            pass
+        else:
+            self.assertTrue(math.isnan(tan))
+
+        try:
+            ntan = math.tan(NINF)
+        except ValueError:
+            pass
+        else:
+            self.assertTrue(math.isnan(ntan))
+
```

Comment:
I concur that this is just a rewrite.  But maybe it's an opportunity to simplify test?

If libm's tan returns nan for non-nan input - the math_1() raises and exception:
https://github.com/python/cpython/blob/d08c7888229e78533648191dfe42e2d2d3ecea25/Modules/mathmodule.c#L932-L936

C stdandards also say if the implementation supports IEEE floating-point arithmetic (IEC 60559): if the argument is ±∞, NaN is returned and [FE_INVALID](https://en.cppreference.com/w/c/numeric/fenv/FE_exceptions) is raised.

```suggestion
        self.assertRaises(ValueError, math.tan, INF)
        self.assertRaises(ValueError, math.tan, NINF)
```

Reply:
What is the history behind the way the test is currently written?

Reply:
This seems to be added in 53876d9cd8, but I don't see yet any glue why.

Reply:
Maybe it *might* be relevant for some alternative implementation, that doesn't require support for IEEE 754 from platform and NaN.  @serhiy-storchaka 

If it worth saving, lets make it at least consistent with other similar tests (testSin/Cos):
```suggestion
        try:
            self.assertTrue(math.isnan(math.tan(INF)))
            self.assertTrue(math.isnan(math.tan(NINF)))
        except ValueError:
            self.assertRaises(ValueError, math.tan, INF)
            self.assertRaises(ValueError, math.tan, NINF)
```

The implementation will return ``nan`` for one infinity and an exception for other - seems to be too odd to accept.

Reply:
Yes, good point.